common/formatting/dynamic.h

```diff
@@ -0,0 +1,361 @@
+#ifndef VERIBLE_COMMON_FORMATTING_DYNAMIC_H_
```

Comment:
If there are concepts and algorithms described in the [rfmt paper](https://github.com/google/rfmt/blob/master/inst/doc/A%20New%20Approach%20to%20Optimal%20Code%20Formatting.pdf) you can refer the reader to, it would be probably good to mention it in the header.